app/services/workflow_errors_reporter.rb

```diff
@@ -0,0 +1,31 @@
+require 'faraday'
+# This class will send a put ReST call to the preservationWF via faraday. This also parses
+# out a shorter error message, that only includes the moab validation error.
+class WorkflowErrorsReporter
+
+  def self.update_workflow(druid, error_message)
+    error_message.select { |error| request(druid, 'moab-valid', moab_error(error)) if error.key?(PreservedObjectHandlerResults::INVALID_MOAB) }
```

Comment:
I see that this is just for the INVALID_MOAB error, mapping to the 'moab-valid' step.   To generalize, would we want a constant in this class of the particular result codes triggering the write to workflow?  Roughly:

in workflow_errors_reporter.rb:
```ruby
ERRORS_TO_REPORT = [
  PreservedObjectHandlerResults::INVALID_MOAB,
  PreservedObjectHandlerResults::PC_PO_VERSION_MISMATCH  # or whatever
].freeze

def self.update_workflow(druid, pohandler_results)
  pohander_results.each do |result| 
     result_code = result.keys.first
     if ERRORS_TO_REPORT.include?(result_code)
        request(druid, map_to_workflow_step(result_code), result.values.first)
    end
end
```

in this case, workflow_errors_reporter is a bit savvy about result codes and result messages, but in a more generic way (not specific to a single error).


OR

you could have the mapping from result_codes to workflow steps in pohandler_result, and from there:

in pohandler_results:

```ruby
def report_results
  report_to_workflows
  ...
end

private

def report_to_workflows
  result_array.each do |result| 
     result_code = result.keys.first
     if MAP_TO_WORKFLOW_STEPS.keys.include?(result_code)
        WorkflowErrorsReporter.update_workflow(druid, MAP_TO_WORKFLOW_STEPS[result_code], workflow_err_msg(results.values.first))
     end
end

def workflow_err_msg(result_value)
  result_value.split(') ').last
end
```

And then WorkflowErrorsReporter could be stupider:  just "i have a workflow step, I have a message, I'm reporting an error."